core/toolbox/toolbox.ts

```diff
@@ -112,10 +112,9 @@ export class Toolbox extends DeleteArea implements IAutoHideable,
     this.workspace_ = workspace;
 
     /** The JSON describing the contents of this toolbox. */
-    // AnyDuringMigration because:  Type 'ToolboxInfo | { contents: never[]; }'
-    // is not assignable to type 'ToolboxInfo'.
-    this.toolboxDef_ = (workspace.options.languageTree || {'contents': []}) as
-        AnyDuringMigration;
+    this.toolboxDef_ =
+        (workspace.options.languageTree ||
+         {'contents': new Array<toolbox.ToolboxItemInfo>()});
```

Comment:
[TypeScript code must not use the Array constructor](https://google.github.io/styleguide/tsguide.html#array-constructor) (even though it seems pretty readable here).  It seems [`: TypeName` is preferred](https://google.github.io/styleguide/tsguide.html#type-assertions-and-object-literals), though I'm not completely clear how to action that advice here.

Also, single quotes around `contents` not needed.

Reply:
As you noted, `: TypeName` doesn't work here; I wasn't able to find any other way to provide type information about an empty, inlined array literal. It might be possible to declare a variable and use that in place of the literal, but since this is only used if workspace.options.languageTree isn't set, I don't love that either. I think the Array constructor is the best bet, particularly since we're not passing any values to it here, so the unexpected behavior issue raised in the styleguide isn't applicable. Happy to change it if you have a specific suggestion/preference though!

Reply:
I think that's a reasonable enough argument, though I expect at some time in the future this will produce a lint error (i.e., when the lint rules get specific enough to enforce the no-`new Array()` rule).